### ISSUE TITLE
[Submission] QuaTorch

### ISSUE THREAD

**Issue by egidioln** [pysimhub/pysimhub.github.io#100]
### Project Name

QuaTorch

### Tagline

Quaternions in PyTorch

### Description

**QuaTorch** is a lightweight python package providing Quaternion, a `torch.Tensor` subclass that represents a [Quaternion](https://en.wikipedia.org/wiki/Quaternion). It implements `torch.compile`-able operations for quaternions such as multiplication, conjugation, inversion, normalization, log, exp, etc. It also supports conversion to/from rotation matrix and axis-angle representation. Convenient utilities are provided together, such as spherical linear interpolation (SLERP) and 3D vector rotation.

### Select all tags that apply

- [ ] block-diagram
- [ ] calibration
- [ ] circuit
- [ ] computational neuroscience
- [ ] control
- [ ] cycle simulation
- [ ] des
- [ ] digital-twin
- [ ] dynamic-systems
- [ ] electromagnetic
- [ ] emulation
- [x] engineering
- [ ] finite-element
- [ ] heating
- [ ] homogenization
- [ ] hydrogen
- [ ] microwave
- [ ] mujoco
- [ ] network
- [ ] neuroscience
- [x] physics
- [ ] powerplant
- [ ] radio-frequency
- [ ] refrigeration
- [ ] rf
- [x] robotics
- [ ] sensitivity analysis
- [x] simulation
- [ ] snn
- [ ] spiking neural network
- [ ] system-modeling
- [ ] thermodynamics
- [ ] transient
- [ ] translation
- [ ] uncertainty quantification
- [ ] vensim
- [ ] visualization
- [ ] xmile

### Additional Tags

quaternions

### GitHub Repository

https://github.com/egidioln/QuaTorch

### Documentation

https://egidioln.github.io/QuaTorch/

### PyPI

https://pypi.org/project/quatorch/

### Example or Tutorial

https://egidioln.github.io/QuaTorch/examples.html

### Logo

https://raw.githubusercontent.com/egidioln/QuaTorch/refs/heads/main/docs/source/_static/logo.svg
